tracer/test/Datadog.Trace.Tools.dd_dotnet.IntegrationTests/Checks/ProcessBasicChecksTests.cs

```diff
@@ -331,16 +333,46 @@ public async Task LdPreloadNotFound()
     }
 
     [SkippableTheory]
-    [InlineData(true)]
-    [InlineData(false)]
-    [InlineData(null)]
-    public async Task DetectContinousProfilerState(bool? enabled)
+    [InlineData("auto", null)]
+    [InlineData("1", null)]
+    [InlineData("0", null)]
+    [InlineData(null, null)]
+    [InlineData("auto", false)]
+    [InlineData("1", false)]
+    [InlineData("0", false)]
+    [InlineData(null, false)]
+    [InlineData("auto", true)]
+    [InlineData("1", true)]
+    [InlineData("0", true)]
+    [InlineData(null, true)]
+    public async Task DetectContinuousProfilerState(string enabled, bool? ssiInjectionEnabled)
     {
         SkipOn.Platform(SkipOn.PlatformValue.MacOs);
-        var environmentVariables = enabled == null ? Array.Empty<(string, string)>()
-            : new[] { ("DD_PROFILING_ENABLED", enabled == true ? "1" : "0") };
+        var ssiInjection = ssiInjectionEnabled is null
+                               ? string.Empty
+                               : ssiInjectionEnabled == true ? "tracer,profiler" : "tracer";
+
+        var expected = (enabled, ssiInjectionEnabled) switch
```

Comment:
Maybe extract this code as a helper method and call it also from WorkingWithContinuousProfiler() instead of duplicating the code

Reply:
Unfortunately easier said than done, as these are in completely different projects etc. We _could_ pull it into a separate file that's linked in both, I'll have a look, thanks 👍 Worst case I'll add a comment